modules/features2d/src/sift.simd.hpp

```diff
@@ -450,31 +450,182 @@ class findScaleSpaceExtremaT
             const sift_wt* currptr = img.ptr<sift_wt>(r);
             const sift_wt* prevptr = prev.ptr<sift_wt>(r);
             const sift_wt* nextptr = next.ptr<sift_wt>(r);
+            int c = SIFT_IMG_BORDER;
 
-            for( int c = SIFT_IMG_BORDER; c < cols-SIFT_IMG_BORDER; c++)
+#if CV_NEON && !(DoG_TYPE_SHORT)
```

Comment:
I propose to use CV_SIMD128 macro to guard the changes. The patch may be relevant for other architectures, besides Intel and ARM.

Reply:
We previously had it using SIMD but the gains were not really showing as worth it so we constrained it.